src/go/wsl-helper/pkg/wsl-utils/version_windows.go

```diff
@@ -282,26 +463,38 @@ func GetWSLInfo(ctx context.Context, log *logrus.Entry) (*WSLInfo, error) {
 	log.Tracef("Got %d appx packages", len(names))
 	for _, name := range names {
 		if version, err := getPackageVersion(name); err == nil {
-			// It seems like the store version _always_ has the kernel,
-			// somewhere; it doesn't seem possible to uninstall it.
 			log.Tracef("Got appx package %s with version %s", name, version)
+			wslVersion, kernelVersion, err := getAppxVersion(ctx, log)
+			if err != nil {
+				return nil, err
+			}
 			return &WSLInfo{
-				Installed: true,
-				Inbox:     false,
-				HasKernel: true,
-				Version:   *version,
+				Installed:      true,
+				Inbox:          false,
+				Version:        *wslVersion,
+				KernelVersion:  *kernelVersion,
+				HasKernel:      PackageVersion{}.Less(*kernelVersion),
+				OutdatedKernel: kernelVersion.Less(MinimumKernelVersion),
 			}, nil
+		} else {
+			log.WithError(err).Trace("Failed to get package version")
 		}
 	}
 
 	log.Trace("Failed to get WSL appx package, trying inbox versions...")
-	hasWSL, hasKernel, err := isInboxWSLInstalled(ctx, log)
+	hasWSL, kernelVersion, err := isInboxWSLInstalled(ctx, log)
```

Comment:
It feels like the function name is somewhat misleading, as it does more than just checking if the inbox version is installed.